index.bs

```diff
@@ -2704,15 +2745,191 @@ partial interface MLGraphBuilder {
+  unsigned long axis = 0;
+};
+
+partial interface MLGraphBuilder {
+  MLOperand gather(MLOperand input, MLOperand indices, optional MLGatherOptions options = {});
+};
+</script>
+
+{{MLGatherOptions}} has the following members:
+<dl dfn-type=dict-member dfn-for=MLGatherOptions>
+    : <dfn>axis</dfn>
+    ::
+        An {{unsigned long}} scalar specifying the axis along which the gathered values are obtained. Its value must be in the range [0, N-1] where N is the [=rank=] of the input tensor.
+</dl>
+
+<div>
+    **Arguments:**
+        - *input*: an {{MLOperand}}. The input N-D tensor from which the values are gathered.
+        - *indices*: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be of type "uint32" in the range [0, N-1] where N is the size of the input dimension indexed by *options.axis*.
+        - *options*: an optional {{MLGatherOptions}}. The optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output N-D tensor of [=rank=] equal to M + N - 1 where M and N are the [=rank=] of the *input* and *indices* tensor respectively.
+</div>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>gather(|input|, |indices|, |options|)</dfn> method steps are:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |input| and |indices| is {{MLOperand}}.
+    1. Let |shapeInput| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} and |rankInput| the [=list/size=] of |shapeInput|.
+    1. Let |shapeIndices| be |indices|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}.
+    1. Let |axis| be |options|.{{MLGatherOptions/axis}}.
+    1. Let |axisSize| be |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}}[|axis|]
+    1. If |axis| is greater than or equal to |rankInput|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. [=map/For each=] |index| &rarr; |value| of |indices|:
+        1. If |index| is greater than or equal to |axisSize|, then [=exception/throw=] a "{{DataError}}" {{DOMException}}.
+    1. Let |dimCount| be zero.
+    1. Let |rankOutput| be zero.
+    1. Let |shapeOutput| be an empty list.
+    1. [=map/For each=] |size| &rarr; |value| of |shapeInput|:
+        1. If |dimCount| is equal to |axis| then [=break=].
+        1. Set |shapeOutput|[|dimCount|] to |size|.
+        1. Increment |dimCount| by one.
+    1. Set |rankOutput| to |dimCount|.
+    1. Let |dimCount| be zero.
+    1. [=map/For each=] |size| &rarr; |value| of |shapeIndices|:
+        1. Set |shapeOutput|[|rankOutput| + |dimCount|] to |size|.
+        1. Increment |dimCount| by one.
+    1. Set |rankOutput| to |rankOutput| + |dimCount|.
+    1. Let |dimCount| be zero.
+    1. [=map/For each=] |size| &rarr; |value| of |shapeInput|:
+        1. If |dimCount| is less than or equal to |axis| then [=continue=].
+        1. Set |shapeOutput|[|rankOutput| + |dimCount|] to |size|.
```

Comment:
I'm implementing baseline implementation, it needs update to

`Set |shapeOutput|[|rankOutput| + |dimCount| - |axis| - 1] to |size|.`

Reply:
Fixed.